frontend/src/App.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import FrontScreen from './Pages/FrontScreen';
+import NavBar from './NavBar';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      "type": null,
+      "latitude": null,
+      "longitude": null
+    }
+    this.setLocation = this.setLocation.bind(this)
+  }
+
+  setType(type) {
+    this.setState({
+      type: type,
+      latitude: this.state.latitude,
+      longitude: this.state.longitude
+    });
+  }
+
+  setLocation() {
+    let microapps = window.microapps;
+    let locationJson;
```

Comment:
you don't need this variable outside, move it inside the .then. 

Limit the scope of variables and declare them where are they assigned/getting used.